src/Models.js

```diff
@@ -1,21 +1,54 @@
-function getExp(skillName){
-  /* Gets exp of a skill from storage.
-   */
+class skill{
```

Comment:
1. Create new module global.js and place there only one variable:
```
const LifeGamification = {};
```
If you want to know why read this: 
https://stackoverflow.com/questions/5063878/how-to-cleanly-deal-with-global-variables
LifeGamification will be your namespaces for globals.

2. Replace all the instances of global variables from
`skillsFullInfo.push(smth);` to `LifeGamification.skillsFullInfo.push(smth);`

3. Wrap the all modules files by anonymous function as follow:
```
(function(){
  // Place the content of the module here. 
})();
```

and then
4. Create 
```class SkillsCollection{
  constructor(skills_array) {
    this.skills_array = skills_array;
  }

  new_skill(name) {
  // Creates new skill to collection
  // TODO implement it fully. 
  }
  save() {
  // Save Collection to the repository
  // It should looks like this
  // TODO implement it fully.
  LifeGamification.repository.save_skills_collection(this);
  }
  get_skill_by_id(id) {
  // This is comment for you not docstring. The name of the function explain everything. 
  // Iterate over all skills which will be placed in the array and return the one with id equal to function argument.
  // Forget about complexity of the algorithm it is not the OI.
  // TODO implement it fully.
  }
  get_skill_by_name(name) {
  // As in the previous funciton.
  // TODO implement it fully.
  }
```
Implement all functions of this class. You can add more function if necessary. 
5. Rename `skill` to `Skill`. The name of the classes should start with Capital letters.
6. In skillRename `nr` to `id`. This name is more meaningfull.
7. Add to the class Skill methods like add_exp() which updates current exp and level.
8. Add at the bottom of the module something like this:
```
LifeGamification.models = {
  skills_collection: new SkillsCollection(LifeGamification.repository.get_skills())
};
```
Ask me if you have any problems.